### ISSUE TITLE
Stuck on "logging in..." until manual page reload [Chrome only?]

### ISSUE THREAD

**Issue by PXCSW-TR** [nextcloud/server#19796]
Hi,

I just upgraded my server from Nextcloud 15 to 17 and now logins get stuck at "logging in..." after providing credentials. But if you manually reload the page, you get logged in and forwarded to your files instantly.

I am aware of this thread:
https://github.com/nextcloud/server/issues/18466

And it seems like the exact same problem. I even have the same error message in chrome, but the solution doesn't work for me and I've had:

  'overwriteprotocol' => 'https',

in my config.php since at least V15 I upgraded to a year ago.

### Steps to reproduce
1. Updated from 15.0.14 -> 16.0.8 -> 17.0.3
2. Access Server in Chrome Browser (Version 80)
3. Input credentials and login
4. Wait.... Forever
5. Manually reload page -> Get transferred to files

### Expected behaviour
Successful login & get forwarded to files at step 4

### Actual behaviour
Stuck on "logging in..."

### Server configuration

**Operating system:** SuSe Enterprise 12.4

**Web server:** apache/2.4.23

**Database:** MariaDB 10.2.29

**PHP version:** PHP 7.2.5

**Nextcloud version:** 17.0.3

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated 15.0.14 -> 16.0.8 -> 17.0.3

**Where did you install Nextcloud from:**
Downloaded the .tar.gz files from https://nextcloud.com/changelog/

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.3.0
  - activity: 2.10.1
  - admin_audit: 1.7.0
  - cloud_federation_api: 1.0.0
  - dav: 1.13.0
  - federatedfilesharing: 1.7.0
  - files: 1.12.0
  - files_rightclick: 0.14.2
  - files_sharing: 1.9.0
  - logreader: 2.2.0
  - lookup_server_connector: 1.5.0
  - oauth2: 1.5.0
  - password_policy: 1.7.0
  - privacy: 1.1.0
  - provisioning_api: 1.7.0
  - serverinfo: 1.7.0
  - sharebymail: 1.7.0
  - text: 1.1.1
  - twofactor_backupcodes: 1.6.0
  - viewer: 1.2.0
  - workflowengine: 1.7.0
Disabled:
  - comments
  - encryption
  - federation
  - files_external
  - files_pdfviewer
  - files_trashbin
  - files_versions
  - files_videoplayer
  - firstrunwizard
  - gallery
  - nextcloud_announcements
  - notifications
  - recommendations
  - support
  - survey_client
  - systemtags
  - theming
  - updatenotification
  - user_ldap

```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
<?php
$CONFIG = array (
  'instanceid' => '**REMOVED**',
  'passwordsalt' => '**REMOVED**',
  'datadirectory' => '/srv/data_share/',
  'dbtype' => 'mysql',
  'version' => '17.0.3.1',
  'dbname' => '**REMOVED**',
  'dbhost' => '127.0.0.1',
  'dbtableprefix' => 'oc_',
  'dbuser' => '**REMOVED**',
  'dbpassword' => '**REMOVED**',
  'installed' => true,
  'default_language' => 'en',
  'forcessl' => false,
  'theme' => 'theme17',
  'skeletondirectory' => '',
  'maintenance' => false,
  'mail_domain' => 'domain.tld',
  'mail_smtpmode' => 'smtp',
  'mail_smtphost' => 'smtp-host.local.domain.tld',
  'mail_smtpport' => '25',
  'logtimezone' => 'Europe/Berlin',
  'trusted_domains' =>
  array (
    0 => 'hostname.domain.tld',
    1 => 'hostname',
    2 => 'hostname.local.domain.tld',
  ),
  'mail_from_address' => '**REMOVED**',
  'secret' => '**REMOVED**',
  'loglevel' => 2,
  'filesystem_check_changes' => 1,
  'updater.release.channel' => 'stable',
  'overwritecondaddr' => '^xxx\\.\\xxx\\xxx\\.xxx$',
  'overwritehost' => 'hostname.domain.tld',
  'overwrite.cli.url' => 'https://hostname.domain.tld',
  'overwriteprotocol' => 'https',
  'simpleSignUpLink.shown' => false,
  'has_internet_connection' => false,
  'trusted_proxies' => ['xxx.xxx.xxx.xxx'],
);

```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LNo

### Client configuration
**Browser:** Google Chrome 80

**Operating system:** Windows 10 / Android 10

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
rev-proxy-ip - - [05/Mar/2020:16:46:12 +0100] "POST /index.php/login HTTP/1.1" 303 - "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36"

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
No entries logged while trying to login
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Refused to send form data to 'http://hostname.domain.tld/' because it violates the following Content Security Policy directive: "form-action 'self'".
```
</details>

**Comment by kesselb** [comment]
Looks like a configuration issue to me. https://help.nextcloud.com/ is the place for such questions. 

1. Make sure the session directory is writable (I guess that works already).
2. Are you sure that you need `overwritecondaddr`? People often use it wrong.
3. It seems that you use a custom theme. Please make sure you updated it accordingly.

**Comment by PXCSW-TR** [comment]
Yes I know about help.nextcloud.com, but I had a similar http / https issue with the password reset function. I posted there and didnt get any replies.

This one: https://help.nextcloud.com/t/unable-to-reset-password-link-and-form-use-http-instead-of-https/58878

But thanks for hinting at the overwritdcondaddr, I actually had no idea what that did.
The reverse proxy config was done by another guy.
I removed the entry from the config.php and now it works fine. Thanks again!

Coincidentally this also fixes my password reset issue I mentioned above.
